source/Nevermore/Advanced/ReadTransaction.cs

```diff
@@ -627,13 +658,17 @@ public Task<DbDataReader> ExecuteReaderAsync(string query, CommandParameterValue
         public DbDataReader ExecuteReader(PreparedCommand preparedCommand)
         {
             using var command = CreateCommand(preparedCommand);
-            return command.ExecuteReader();
+            var result = command.ExecuteReader();
+            ExecutedCommands.Add(preparedCommand);
```

Comment:
I thought it is worth re-executing commands issued through the `ExecuteReader` and `ExecuteScalar` methods because there might be `INSERT` and `UPDATE` statements on them. For example:

```
UPDATE FROM table SET ... WHERE ...;
SELECT * FROM table;
```